astropy/modeling/parameters.py

```diff
@@ -191,7 +193,7 @@ class Parameter:
 
     def __init__(self, name='', description='', default=None, unit=None,
                  getter=None, setter=None, fixed=False, tied=False, min=None,
-                 max=None, bounds=None, prior=None, posterior=None):
+                 max=None, bounds=None, prior=None, posterior=None, mag=False):
```

Comment:
Is it necessary to add a new argument? I would think that for most models, magnitudes will just fail miserably if passed in. I.e., perhaps the check in `_set_unit` is not necessary? (But no real idea about the modelling API, so feel free to ignore!!)

Reply:
Models which involve a multiply/divide of a magnitude parameter value fail in a horribly unintuitive way; moreover, we cannot support `MagUnit` type units in general way. This sets it up so that `MagUnit` type units will not be allowed in a parameter unless the model designer specifically allows them. The goal here is to allow `MagUnit` units in appropriate parameters while preventing uninformative tracebacks.

Reply:
OK, if the failure is hopeless unintuitive, then something to guard the user seems well worth it.